src/components/reusable/blockCodeView/blockCodeView.ts

```diff
@@ -0,0 +1,520 @@
+   * @internal
+   */
+  @state()
+  private _copyState = { copied: false, text: '' };
+
+  /** Detected language for the code snippet.
+   * @internal
+   */
+  @state()
+  private _effectiveLanguage = '';
+
+  /** Highlighted code to be displayed.
+   * @internal
+   */
+  @state()
+  private _highlightedCode = '';
+
+  /** Code snippet fits into the height of the container -- no expansion needed.
+   * @internal
+   */
+  @state()
+  private _codeFitsContainerOnLoad = true;
+
+  /** If expandable -- height of the container when fully expanded.
+   * @internal
+   */
+  @state()
+  private _expandedHeight: number | null = null;
+
+  override render() {
+    const containerStyle = this.getContainerStyle();
+
+    return html`
+      ${
+        this.codeViewLabel
+          ? html`<div class="code-view__label">
+              <label>${this.codeViewLabel}</label>
+            </div>`
+          : null
+      }
+      <div
+        aria-label=${ifDefined(this.ariaLabelAttr)}
+        class="${this.getContainerClasses()}"
+      >
+          <div class="code-snippet-wrapper" style=${containerStyle}>
+              <pre
+              @keydown=${this.handleKeypress}
+              role="region"
+            ><code tabindex="0" class="language-${
+              this._effectiveLanguage
+            }">${unsafeHTML(this._highlightedCode)}</code></pre>
+          </div>
+
+          ${
+            this.copyOptionVisible
+              ? html`<kd-button
+                  class="code-view__copy-button"
+                  kind="tertiary"
+                  size="small"
+                  iconPosition="left"
+                  ?disabled=${this._copyState.copied}
+                  title=${ifDefined(this.copyButtonTitleAttr)}
+                  name=${ifDefined(this.copyButtonDescriptionAttr)}
+                  description=${ifDefined(this.copyButtonDescriptionAttr)}
+                  @click=${this.copyCode}
+                >
+                  <kd-icon
+                    slot="icon"
+                    class="copy-icon"
+                    .icon=${this._copyState.copied ? checkmarkIcon : copyIcon}
+                  ></kd-icon>
+                  ${this._copyState.text
+                    ? html`<span class="copy-text"
+                        >${this._copyState.text}</span
+                      >`
+                    : null}
+                </kd-button>`
+              : null
+          }
+          ${
+            this.codeViewExpandable &&
+            (this.hasOverflow || !this._codeFitsContainerOnLoad)
+              ? html`<kd-button
+                  class="code-view__expand-button"
+                  kind="tertiary"
+                  size="small"
+                  iconPosition="left"
+                  title="Expand/Collapse code snippet"
+                  name="toggle-code-expanded"
+                  description="Click to ${this.codeExpanded
+                    ? 'collapse'
+                    : 'expand'} the full code snippet"
```

Comment:
Can't hard-code strings. They must be configurable for localization purposes. Take a look at how I've managed this in radio button group for example using deepmerge on the `textStrings` prop.

Reply:
✅ -- mirrored your `textStrings` implementation